credential/cargo-credential/tests/examples.rs

```diff
@@ -0,0 +1,23 @@
+use snapbox::cmd::Command;
+
+#[test]
+fn file_provider() {
+    let bin = snapbox::cmd::compile_example("file-provider", []).unwrap();
+
+    let hello = r#"{"v":[1]}"#;
+    let login_request = r#"{"v": 1,"registry": {"index-url":"https://github.com/rust-lang/crates.io-index","name":"crates-io"},"kind": "login","token": "s3krit","args": []}"#;
+    let login_response = r#"{"Ok":{"kind":"login"}}"#;
+
+    let get_request = r#"{"v": 1,"registry": {"index-url":"https://github.com/rust-lang/crates.io-index","name":"crates-io"},"kind": "get","operation": "read","args": []}"#;
+    let get_response =
+        r#"{"Ok":{"kind":"get","token":"s3krit","cache":"session","operation_independent":true}}"#;
+
+    Command::new(bin)
+        .current_dir(env!("CARGO_TARGET_TMPDIR"))
```

Comment:
This is the same `CARGO_TARGET_TMPDIR` used by every test in every package in this workspace.  Should we add something to this to make it more unique / descriptive?

Reply:
Added `/cargo-credential-tests`, and automatic cleanup.